beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -440,11 +479,46 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         packages: Vec<GossipAggregatePackage<T::EthSpec>>,
         reprocess_tx: Option<mpsc::Sender<ReprocessQueueMessage>>,
     ) {
-        let aggregates = packages.iter().map(|package| package.aggregate.as_ref());
+        let (to_process, skip): (
+            Vec<(
+                GossipAggregatePackage<T::EthSpec>,
+                SignedAggregateAndProof<T::EthSpec>,
+            )>,
+            Vec<(AttnError, GossipAggregatePackage<T::EthSpec>)>,
+        ) = packages.into_iter().partition_map(|p| {
+            match self
+                .chain
+                .is_lazy_att_observed_subset(&p.aggregate.message.aggregate)
+            {
+                Ok(false) => match p.aggregate.clone().not_lazy() {
+                    Ok(agg) => itertools::Either::Left((p, agg)),
+                    Err(e) => itertools::Either::Right((e.into(), p)),
+                },
+                Ok(true) => itertools::Either::Right((
+                    AttnError::AttestationSupersetKnown(
+                        p.aggregate.message.aggregate.data.tree_hash_root(),
```

Comment:
If we pass the attestation data root to `is_lazy_att_observed_subset` above we can avoid re-computing it here, and save a few milliseconds